crates/ruff/src/autofix/mod.rs

```diff
@@ -65,6 +66,15 @@ fn apply_fixes<'a>(
             continue;
         }
 
+        // If this fix requires isolation, and we've already applied another fix that
+        // requires isolation, skip it. We apply at most one isolated fix per run.
```

Comment:
Oh, this is interesting. I expected that we run `isolated` fixes in full isolation -> We apply no other fixes in that pass. But that's not what we do. Is there another word that may describe this behavior better? 

I guess I expected `Serializable` isolation (using SQL terminology) and our default isolation level is `NO_OVERLAPPING`. The way I think about this new isolation level is similar to NON_CONCURRENT but that feels wrong too. Hmm

On naming:
* Should we rename `Isolation` to `IsolationLevel`
* Should we rename the default isolation to `NoOverlapping` (or similar) because all fixes have an isolation.